flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/operators/route/RouteFunction.java

```diff
@@ -0,0 +1,171 @@
+import com.ververica.cdc.common.event.RenameColumnEvent;
+import com.ververica.cdc.common.event.SchemaChangeEvent;
+import com.ververica.cdc.common.event.TableId;
+import com.ververica.cdc.common.schema.Selectors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.ververica.cdc.common.utils.Preconditions.checkState;
+
+/** A map function that applies user-defined routing logics. */
+public class RouteFunction extends RichMapFunction<Event, Event> {
+    private final List<Tuple2<String, TableId>> routingRules;
+    private transient List<Tuple2<Selectors, TableId>> routes;
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /** Builder of {@link RouteFunction}. */
+    public static class Builder {
+        private final List<Tuple2<String, TableId>> routingRules = new ArrayList<>();
+
+        public Builder addRoute(String tableInclusions, TableId replaceBy) {
+            routingRules.add(Tuple2.of(tableInclusions, replaceBy));
+            return this;
+        }
+
+        public RouteFunction build() {
+            return new RouteFunction(routingRules);
+        }
+    }
+
+    private RouteFunction(List<Tuple2<String, TableId>> routingRules) {
+        this.routingRules = routingRules;
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        routes =
+                routingRules.stream()
+                        .map(
+                                tuple2 -> {
+                                    String tableInclusions = tuple2.f0;
+                                    TableId replaceBy = tuple2.f1;
+                                    Selectors selectors =
+                                            new Selectors.SelectorsBuilder()
+                                                    .includeTables(tableInclusions)
+                                                    .build();
+                                    return new Tuple2<>(selectors, replaceBy);
+                                })
+                        .collect(Collectors.toList());
+    }
+
+    @Override
+    public Event map(Event event) throws Exception {
+        checkState(
+                event instanceof ChangeEvent,
+                String.format(
+                        "The input event of the route is not a ChangeEvent but with type \"%s\"",
+                        event.getClass().getCanonicalName()));
+        ChangeEvent changeEvent = (ChangeEvent) event;
+        TableId tableId = changeEvent.tableId();
+
+        for (Tuple2<Selectors, TableId> route : routes) {
+            Selectors selectors = route.f0;
+            TableId replaceBy = route.f1;
+            if (selectors.isMatch(tableId)) {
+                return recreateChangeEvent(changeEvent, replaceBy);
+            }
+        }
+        return event;
+    }
+
+    private ChangeEvent recreateChangeEvent(ChangeEvent event, TableId tableId) {
```

Comment:
Can we add a `copy` method to ChangeEvent to avoid these complex classifications？

Reply:
Hmmm my concern is that it might be confusing if we introduce this `copy` in the `ChangeEvent`, which is now a clean interface. 

`copy` sounds like to create a new event which is exactly the same, but here we need to rewrite the table ID field, that means something like `replaceTableId(TableId)`, but it's quite confusing to implementations that why they have to provide an ability to change table id only but not for other fields.